packages/openapi3/src/openapi.ts

```diff
@@ -306,6 +306,7 @@ function createOAPIEmitter(
       } as any,
       context
     );
+    diagnostics = createDiagnosticCollector();
```

Comment:
can we move it to line 213?

Reply:
no this needs to be reset for each service, but could move line 289 I think